plugins/inputs/win_services/win_services_test.go

```diff
@@ -0,0 +1,101 @@
+// +build windows
+
+//this test must be run under administrator account
+package win_services
+
+import (
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"testing"
+    "golang.org/x/sys/windows/svc/mgr"
+)
+
+var InvalidServices = []string{"XYZ1@", "ZYZ@", "SDF_@#"}
+var KnownServices = []string{"LanmanServer", "TermService"}
+
+func TestList(t *testing.T) {
+	services, err := listServices(KnownServices)
```

Comment:
I'm not sure what is the best goal here.

I can easily mock listServices to test Gather, but to properly test listServices I will need to mock Mgr and Service from mgr package. As there is type dependency, as Mgr.OpenService returns mgr.Service, I can not just create an interface which will be satisfied by existing structs in mgr package and their methods.
I would need to create proxies of both to satisfy new interface in real implementation and this will complicate the code. Did you mean this?